pkg/connector/account.go

```diff
@@ -276,13 +286,108 @@ func (o *accountResourceType) Grant(ctx context.Context, principal *v2.Resource,
 		TargetType:       awsSsoAdminTypes.TargetTypeAwsAccount,
 	}
 
-	if _, err := o.ssoAdminClient.CreateAccountAssignment(ctx, inp); err != nil {
+	createOut, err := o.ssoAdminClient.CreateAccountAssignment(ctx, inp)
+	if err != nil {
 		return nil, err
 	}
 
+	attemptCount := 1
+	attemptLogger := l.With(
+		zap.Int("attempt", attemptCount),
+		zap.String("request_id", awsSdk.ToString(createOut.AccountAssignmentCreationStatus.RequestId)),
+		zap.String("principal_id", awsSdk.ToString(createOut.AccountAssignmentCreationStatus.PrincipalId)),
+		zap.String("principal_type", string(createOut.AccountAssignmentCreationStatus.PrincipalType)),
+		zap.String("permission_set_arn", awsSdk.ToString(createOut.AccountAssignmentCreationStatus.PermissionSetArn)),
+	)
+
+	complete, err := o.checkCreateAccountAssignmentStatus(ctx, attemptLogger, createOut.AccountAssignmentCreationStatus)
+	if err != nil {
+		var ae *awsSsoAdminTypes.AccessDeniedException
+		if errors.As(err, &ae) {
+			l.Info("aws-connector: access denied while attempting to check status. Assuming account assignment creation is complete.", zap.Error(err))
+			complete = true
+		} else {
+			return nil, err
+		}
+	}
+
+	waitCtx, cancel := context.WithTimeout(ctx, AccountAssignmentMaxWaitDuration)
+	defer cancel()
+
+	for !complete {
+		select {
+		case <-waitCtx.Done():
+			return nil, fmt.Errorf("aws-connector: account assignment creation timed out: %w", ctx.Err())
+		case <-time.After(AccountAssignmentRetryDelay):
+		}
+
+		attemptCount++
+		attemptLogger.Debug("aws-connector: waiting for account assignment creation to complete, checking status...")
+		complete, err = o.checkCreateAccountAssignmentStatus(waitCtx, attemptLogger, createOut.AccountAssignmentCreationStatus)
```

Comment:
Can we actually use `complete` to determine the `grant` status? If grant failed, completed is true, does the UI update properly that grant failed?

Reply:
if it fails, completed is true, but we also return an error indicating that it failed, so we'd return immediately.